plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergTableHandle.java

```diff
@@ -80,6 +81,7 @@ public IcebergTableHandle(
             @JsonProperty("tableLocation") String tableLocation,
             @JsonProperty("storageProperties") Map<String, String> storageProperties,
             @JsonProperty("retryMode") RetryMode retryMode,
+            @JsonProperty("isUpdateOrDelete") boolean isUpdateOrDelete,
```

Comment:
Sounds like something engine could (should?) provide directly to `ConnectorSplitManager`.

cc @losipiuk